cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -68,20 +69,21 @@ func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) erro
 	}
 
 	// User admin should have full access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "admin", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for admin: %v", err)
+	if err := createKubeConfigFileForClient(masterEndpoint, AdminKubeConfigFileName, []string{"system:masters"}, outDir, caCert, caKey); err != nil {
+		return fmt.Errorf("couldn't create %s: %v", AdminKubeConfigFileName, err)
 	}
 
-	// TODO: The kubelet should have limited access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "kubelet", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for kubelet: %v", err)
+	// The kubelet should have limited access to the cluster
+	if err := createKubeConfigFileForClient(masterEndpoint, KubeletKubeConfigFileName, []string{"system:nodes"}, outDir, caCert, caKey); err != nil {
+		return fmt.Errorf("couldn't create %s: %v", KubeletKubeConfigFileName, err)
 	}
 
 	return nil
 }
 
-func createKubeConfigFileForClient(masterEndpoint, client, outDir string, caCert *x509.Certificate, caKey *rsa.PrivateKey) error {
-	key, cert, err := certphase.NewClientKeyAndCert(caCert, caKey)
+func createKubeConfigFileForClient(masterEndpoint, kubeConfigFileName string, organization []string, outDir string, caCert *x509.Certificate, caKey *rsa.PrivateKey) error {
+	client := strings.Split(kubeConfigFileName, ".")[0]
```

Comment:
agree. you already have an arg for the outputdir... just change that to be the output filepath instead of building the output filepath inside this function